### ISSUE TITLE
State Provider

### ISSUE THREAD

**Issue by werne** [vitmalina/w2ui#517]
It would make the applications quite user friendly if w2ui would have some sort of State Provider API (like http://docs.sencha.com/extjs/3.4.0/#!/api/Ext.state.Provider ) and support in the components, that when configured, to auto save/reload their settings.

E.g. Cookie Provider, Local Storage Provider, etc. so that info as: layout resizing, column resizing, default sort order, tree node collaps, etc. could be persisted/reloaded automatically.

tnx.

**Comment by gpetrov** [comment]
Please add this also for the Layout - so it we can save the last user resized layout and restore it on next load

**Comment by aadrian** [comment]
+1
This would be required for Layout, Sidebar, Tabs and popup too.

**Comment by hansgru** [comment]
Yes, indeed, the sidebar (tree) also needs this, and for Tabbed applications have the Tabs also remember this would be of huge help.

**Comment by gpetrov** [comment]
@vitmalina any chance for reopening this and implementing it? It is  great feature and everybody needs it!

**Comment by vitmalina** [comment]
Layout and sidebar - maybe. But toolbar and popup - no. Tabs - unsure.

**Comment by gpetrov** [comment]
@vitmalina  any news on implementing this?

**Comment by vitmalina** [comment]
No updates on this. Sorry. I am working on items with higher priority.
